kmir/src/kmir/kmir.py

```diff
@@ -55,6 +57,96 @@ def __init__(
         KParse.__init__(self, definition_dir)
         self.llvm_library_dir = llvm_library_dir
 
+    @staticmethod
+    def from_kompiled_kore(
+        smir_info: SMIRInfo, bug_report: Path | None = None, symbolic: bool = True, target_dir: str = 'out-kore'
+    ) -> KMIR:
+        kmir = KMIR(HASKELL_DEF_DIR)
+
+        def _insert_rules_and_write(input_file: Path, rules: list[str], output_file: Path) -> None:
+            with open(input_file, 'r') as f:
+                lines = f.readlines()
+
+            # last line must start with 'endmodule'
+            last_line = lines[-1]
+            assert last_line.startswith('endmodule')
+            new_lines = lines[:-1]
+
+            # Insert rules before the endmodule line
+            new_lines.append(f'\n// Generated from file {input_file}\n\n')
+            new_lines.extend(rules)
+            new_lines.append('\n' + last_line)
+
+            # Write to output file
+            with open(output_file, 'w') as f:
+                f.writelines(new_lines)
+
+        target_path = Path(target_dir)
+        # TODO if target dir exists and contains files, check file dates (definition files and interpreter)
+        # to decide whether or not to recompile. For now we always recompile.
+        target_path.mkdir(parents=True, exist_ok=True)
+
+        rules = kmir.make_kore_rules(smir_info)
+        _LOGGER.info(f'Generated {len(rules)} function equations to add to `definition.kore')
+
+        if not symbolic:
+            raise Exception('Non-symbolic (LLVM) kore-module not implemented')
+
+        # Create output directories
+        target_llvm_path = target_path / 'llvm-library'
+        target_llvmdt_path = target_llvm_path / 'dt'
+        target_hs_path = target_path / 'haskell'
+
+        _LOGGER.info(f'Creating directories {target_llvmdt_path} and {target_hs_path}')
+        target_llvmdt_path.mkdir(parents=True, exist_ok=True)
+        target_hs_path.mkdir(parents=True, exist_ok=True)
+
+        # Process LLVM definition
+        _LOGGER.info('Writing LLVM definition file')
+        llvm_def_file = LLVM_LIB_DIR / 'definition.kore'
+        llvm_def_output = target_llvm_path / 'definition.kore'
+        _insert_rules_and_write(llvm_def_file, rules, llvm_def_output)
+
+        # Run llvm-kompile-matching and llvm-kompile for LLVM
+        # TODO use pyk to do this if possible (subprocess wrapper, maybe llvm-kompile itself?)
+        # TODO align compilation options to what we use in plugin.py
```

Comment:
There certainly is a better way to run `llvm-kompile-matching` and `llvm-kompile` ?